rfcs/0000-switch-postgresql-from-ec2-to-rds.md

```diff
@@ -0,0 +1,190 @@
+For a team as small as ours, this counts as a rather significant win:
+  we're freeing our time and energy up to focus on more important things.
+
+One of the most important considerations here is cost: RDS isn't free.
+For details, please see the attached
+  [RFC postgresql-on-rds Addendum: AWS Pricing](https://docs.google.com/spreadsheets/d/1eeMVqy9xdctcJO80M1JGVqd6apUaza-hfKGtIKLhIdE/edit?usp=sharing)
+  spreadsheet, but at a high-level:
+
+* RDS Aurora charges about twice as much for compute.
+* RDS Aurora does not charge for extra storage for each compute instance; all instances within a region share the same storage.
+    * As storage-per-instance is currently project to be our largest and fastest-growing AWS expense,
+      this would provide us with very large savings.
+* RDS Aurora also introduces a new charge of $0.20 per 1M I/O operations (both read & write).
+  With our projected number of DB queries, this will not be a major cost driver.
+    * It could _become_ a major cost driver if our usage wildly exceeds projections.
+      But, in that scenario, having the more reliable managed DB-as-a-service backing us becomes an even bigger concern that justifies the cost.
+
+Overall, RDS Aurora will likely save us anywhere from 27% to 71% on our AWS costs, through 2020-06: it's a major win on cost control.
+
+It's worth mentioning that we're actually already leveraging RDS for several things:
+
+* RDS has powered the "old sandbox" all along.
+* RDS was also used as the database for the initial load development and benchmarking that was done
+
+This change is an "under the hood" thing only; it does not directly impact external users of our systems.
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+For the most important pieces of our systems, the Data Pipeline and Data Server,
+  RDS Aurora is a fully-compatible switch: all they need to change are their JDBC URLs.
+For our infrastructure automation, RDS will be a major win, as we'll go from having to orchestrate an EC2 instance,
+  operating system, _and_ PostgreSQL configuration to just asking AWS to, "give us an RDS instance, please."
+
+Once ready, The deployment/transition to RDS Aurora should be simple and smooth:
+
+1. Update our infrastructure playbooks to deploy and configure the desired RDS Aurora instances.
+    * Should also include duplicate Data Pipeline and Data Server instances,
+      connected to the new RDS Aurora instances.
+2. Pause all ETL (i.e. the RIF export and the Data Pipeline service) in the environment being transitioned.
+3. Fork the RIF export into a second S3 location, for the new Data Pipeline service.
+4. Restore a current backup of the environment's DB to the primary RDS Aurora instance.
+   Wait for the restore to complete and for it to replicate to all secondary instances.
+5. Resume all ETL.
```

Comment:
On PostgreSQL the restore from backup took about 4 days. The setup replicas took about 24 additional hours. Aurora may be faster, but we don't know. Suggest that we do a trial run, this restore to Aurora first without turning ETL off to learn how fast the restore will take.